### PR TITLE
`Programming exercises`: Fix task in problem statement with zero tests

### PR DESCRIPTION
**Pull request by JohannesStoehr**
#### General

- [x] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] I chose a title conforming to the [naming conventions for pull requests](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/development-process.html#naming-conventions-for-github-pull-requests).
#### Client
- [x] I followed the [coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client/) and ensured that the layout is responsive.
- [x] I added multiple integration tests (Jest) related to the features (with a high test coverage), while following the [test guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client-tests/).
- [x] I added multiple screenshots/screencasts of my UI changes.
- [x] I translated all newly inserted strings into English and German.

### Motivation and Context

Sometimes a task in a problem statement does not contain any tests. This can be the case if the exercise is only/partially graded manually, but the tasks are still used for structuring.
Currently such a task would still show "0 of 1 test passing".

### Description

This PR fixes this behavior such that "No tests" is displayed instead.
In case a hidden test is added to the task before the due date, the task will still show "0 of 1 test passing", this is out of scope for this PR.

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Students
- 1 Programming Exercise

1. Log in to Artemis
2. Create a task in the problem statement without any tests or delete the tests from an existing task
3. Participate as a student and submit something
4. Look at the task and klick on the feedback to verify that there is no misleading text for this case

#### Code Review
- [ ] Review 1
- [x] Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

### Screenshots

Before:
Bildschirm­foto 2022-12-16 um 19 15 41
After:
Bildschirm­foto 2022-12-16 um 19 16 24

### CODE REVIEW

**Review by timo07k — DISMISSED** [review]
Works exactly as shown in the screenshot. Tested on TS1.

**Review by PhilipWerz — DISMISSED** [review]
Tested removing tests from tasks as well as creating new tasks without tests multiple times. Everything worked as it should. Tested on TS1

File: src/test/javascript/spec/component/programming-exercise/programming-exercise-instruction.component.spec.ts [hunk 1]
```diff
@@ -479,4 +477,89 @@ describe('ProgrammingExerciseInstructionComponent', () => {
             },
         });
     }));
+
+    it('foo (non legacy case)', fakeAsync(() => {
```
Comment (Strohgelaender) [inline src/test/javascript/spec/component/programming-exercise/programming-exercise-instruction.component.spec.ts:481]: Can you give this test case a better name?

**Review by Strohgelaender — COMMENTED** [review]
code looks good, only one small comment:

File: src/test/javascript/spec/component/programming-exercise/programming-exercise-instruction.component.spec.ts (hunk 1 above)
Comment (JohannesStoehr) [reply src/test/javascript/spec/component/programming-exercise/programming-exercise-instruction.component.spec.ts:481]: Oops, solved

**Review by Strohgelaender — DISMISSED** [review]

**Review by simon-hng — DISMISSED** [review]
Code

**Review by tobias-lippert — CHANGES_REQUESTED** [review]
Tested on ts2.
When I click on the result, the test case that has been removed is still shown, although the task shows no tests.

**Review by laadvo — COMMENTED** [review]
Tasks with no tests desplay "No tests" before starting the exercise and after the test have already been run as expected. You can however still access the feedback of the nonexistent test by clicking the "?" in the tasks overview.
image. 
For now it says "No result details available.", this could maybe also be changed to "No tests"

**Review by tobias-lippert — APPROVED** [review]
Approve after discussion on slack that showing all tests is intended behavior.

**Review by laadvo — APPROVED** [review]
The tasks without tests working as expected, you cannot access the feedback by clicking on the task in the task bar any longer. Tested on ts2.

File: src/main/webapp/app/exercises/programming/shared/instructions-render/service/programming-exercise-instruction.service.ts [hunk 2]
```diff
@@ -40,7 +40,7 @@ export class ProgrammingExerciseInstructionService {
             const { failed, notExecuted, successful } = this.separateTests(tests, latestResult);
 
             // Exercise is done if none of the tests failed
-            const testCaseState = failed.length > 0 ? TestCaseState.FAIL : notExecuted.length > 0 ? TestCaseState.NOT_EXECUTED : TestCaseState.SUCCESS;
+            const testCaseState = failed.length > 0 ? TestCaseState.FAIL : notExecuted.length > 0 || tests.length === 0 ? TestCaseState.NOT_EXECUTED : TestCaseState.SUCCESS;
```
Comment (Strohgelaender) [inline src/main/webapp/app/exercises/programming/shared/instructions-render/service/programming-exercise-instruction.service.ts:43]: This line is quite long, can you make this an if/else construct?

**Review by simon-hng — APPROVED** [review]
Code

**Review by stefanwaldhauser — APPROVED** [review]
Nice fix! Works as expected

**Review by Strohgelaender — APPROVED** [review]

**Pull request merged**